tests/conftest.py

```diff
@@ -5,7 +5,494 @@
 # import path, so we construct one.
 # todo: reorg core file structure so we have a reliable package name
 from os.path import abspath, dirname, basename
+import random
+import uuid
+
+import pytest
+from sqlalchemy import create_engine
+from sqlalchemy.exc import ProgrammingError
+from sqlalchemy.orm.session import Session
+
+from core.classifier import Classifier
+from core.config import Configuration
+from core.lane import Lane
+
+from core.model import Base, classifier, get_one_or_create
+from core.model.admin import Admin
+from core.model.classification import Genre, Subject
+from core.model.collection import Collection
+from core.model.configuration import ExternalIntegration
+from core.model.constants import MediaTypes
+from core.model.contributor import Contributor
+from core.model.datasource import DataSource
+from core.model.edition import Edition
+from core.model.identifier import Identifier
+from core.model.library import Library
+from core.model.licensing import DeliveryMechanism, LicensePool, LicensePoolDeliveryMechanism, RightsStatus
+from core.model.resource import Hyperlink, Representation
+from core.model.work import Work
+from core.util.datetime_helpers import utc_now
 
 # Pull in the session_fixture defined in core/testing.py
 # which does the database setup and initialization
 pytest_plugins = ["{}.testing".format(basename(dirname(dirname(abspath(__file__)))))]
+
+
+@pytest.fixture(autouse=True, scope="session")
+def init_test_db():
+    db_url = Configuration.database_url()
+    engine = create_engine(db_url)
+    for table in reversed(Base.metadata.sorted_tables):
+        try:
+            engine.execute(table.delete())
+        except ProgrammingError:
+            pass
+
+    with engine.connect() as conn:
+        Base.metadata.create_all(conn)
+
+    engine.dispose()
+
+
+@pytest.fixture
+def db_engine():
+    db_url = Configuration.database_url()
+    engine = create_engine(db_url)
+    yield engine
+    engine.dispose()
+
+
+@pytest.fixture
+def db_session(db_engine):
+    with db_engine.connect() as connection:
+        transaction = connection.begin_nested()
+        session = Session(connection)
+        yield session
+        transaction.rollback()
```

Comment:
I know this seems like the right way to do this, but I have to say some part of me doesn't entirely trust it. Maybe peek at the db after a test run some time just to see if there's anything floating around in there that shouldn't have stuck around between tests.